### PR TITLE
Unit test nfs improvements

### PR DESCRIPTION
**Pull request by tasleson**
* Extend the unit tests for nfs and fix an issue
* Clean-up misc. issues along the way

resolves https://github.com/open-iscsi/targetd/issues/65

### CODE REVIEW

**Comment by tasleson** [comment]
@dsonck92 If you have some time I would appreciate having you looking this over, thanks!

**Review by dsonck92 — CHANGES_REQUESTED** [review]
Apart from the small PEP8 nitpick it looks good to me. It definitely has some more coverage and I also saw some nice code simplification (conflicting options check).

File: test/targetd_test.py [hunk 1]
```diff
@@ -703,10 +747,27 @@ def test_gp_nfs_export_add(self):
...
             self._fs_destroy(fs)
 
+    def test_ep_nfs_export_invalid_options(self):
+        for fs_pool in TestTargetd._fs_pools():
+            fs = TestTargetd._fs_create(fs_pool, rs(length=10))
+
+            error_code = 0
+            try:
+                self._nfs_export_add("0.0.0.0/0", fs.full_path,
+                                          ["insecure", "secure"], "1000")
```
[the remark this answers is not in the document]
Comment (tasleson) [reply test/targetd_test.py]: Done, thanks!

**Review by dsonck92 — APPROVED** [review]

**Comment by tasleson** [comment]
@dsonck92 Thanks for reviewing!

**Pull request merged**